pkg/config/validation/validation.go

```diff
@@ -2786,7 +2787,9 @@ func validateHTTPRouteDestinations(weights []*networking.HTTPRouteDestination) (
 			errs = appendErrors(errs, ValidateHTTPHeaderOperationName(name))
 		}
 
-		errs = appendErrors(errs, validateDestination(weight.Destination))
+		if !gatewaySemantics {
+			errs = appendErrors(errs, validateDestination(weight.Destination))
```

Comment:
nit: we could just allow `""` for gatewaySemantics, since validateDestination has some other checks that are still useful